azurerm/resource_arm_iothub.go

```diff
@@ -332,3 +564,148 @@ func flattenIoTHubSharedAccessPolicy(input *[]devices.SharedAccessSignatureAutho
 
 	return results
 }
+
+func flattenIoTHubEndpoint(input *devices.RoutingProperties) []interface{} {
+	results := make([]interface{}, 0)
+
+	if input != nil && input.Endpoints != nil {
+
+		for _, container := range *input.Endpoints.StorageContainers {
+			output := make(map[string]interface{}, 0)
+
+			if connString := container.ConnectionString; connString != nil {
+				output["connection_string"] = *connString
+			}
+			if name := container.Name; name != nil {
+				output["name"] = *name
+			}
+			if containerName := container.ContainerName; containerName != nil {
+				output["container_name"] = *containerName
+			}
+			if fileNameFmt := container.FileNameFormat; fileNameFmt != nil {
+				output["file_name_format"] = *fileNameFmt
+			}
+			if batchFreq := container.BatchFrequencyInSeconds; batchFreq != nil {
+				output["batch_frequency_in_seconds"] = *batchFreq
+			}
+			if chunkSize := container.MaxChunkSizeInBytes; chunkSize != nil {
+				output["max_chunk_size_in_bytes"] = *chunkSize
+			}
+			if encoding := container.Encoding; encoding != nil {
+				output["encoding"] = *encoding
+			}
+
+			results = append(results, output)
+		}
+
+		for _, queue := range *input.Endpoints.ServiceBusQueues {
+			output := make(map[string]interface{}, 0)
+
+			if connString := queue.ConnectionString; connString != nil {
+				output["connection_string"] = *connString
+			}
+			if name := queue.Name; name != nil {
+				output["name"] = *name
+			}
+
+			results = append(results, output)
+		}
+
+		for _, topic := range *input.Endpoints.ServiceBusTopics {
```

Comment:
can we nil-check around this to ensure `input.Endpoints.ServiceBusTopics ` isn't nil?